docs/contributing/index.rst

```diff
@@ -338,70 +602,109 @@ A good etiquette to take over is:
-    to geomstats is to pick an item from the list of `starter project
-    <https://github.com/geomstats/geomstats/labels/starter%project>`_ in the issue
-    tracker. Your assistance in this area will be greatly appreciated by the
-    more experienced developers as it helps free up their time to concentrate on
-    other issues.
+3. Use understandable function and variable names. The naming should help
+   the maintainers reading faster through your code. Thus, ``my_array``,
+   ``aaa``, ``result``, ``res`` are generally bad variable names,
+   whereas ``rotation_vec`` or ``symmetric_mat`` read well.
 
-.. topic:: help wanted tag
+4. Avoid comment in the code, the documentation goes in the docstrings.
+   This allows the explanations to be included in the documentation
+   generated automatically on the website. Furthermore, forbidding comments
+   forces us to write clean code, and clean docstrings.
 
-    We often use the help wanted tag to mark issues regardless of difficulty. Additionally,
-    we use the help wanted tag to mark Pull Requests which have been abandoned
-    by their original contributor and are available for someone to pick up where the original
-    contributor left off. The list of issues with the help wanted tag can be found
-    `here <https://github.com/geomstats/geomstats/labels/help%20wanted>`__ .
+5. Follow geomstats' API. For example, points on manifolds are denoted
+   ``point``, tangent vectors ``tangent_vec``, matrices ``mat``, exponential
+   ``exp`` and logarithms ``log``.
 
-    Note that not all issues which need contributors will have this tag.
+6. Avoid multiple statements on one line. Divide complex computations on several
+   lines. Prefer a line return after a control flow statement (``if``/``for``).
+
+7. **Don't use** ``import *`` **in any case**. It is considered harmful
+   by the `official Python recommendations
+   <https://docs.python.org/2/howto/doanddont.html#from-module-import>`_.
+   It makes the code harder to read as the origin of symbols is no
+   longer explicitly referenced, but most important, it prevents
+   using a static analysis tool like `pyflakes
+   <https://divmod.readthedocs.io/en/latest/products/pyflakes.html>`_ to
+   automatically find bugs in geomstats.
+
+8. Avoid the use of ``import ... as`` and of ``from ... import foo, bar``,
+   i.e. do not rename modules or modules' functions, because you would create
+   objects living in several namespaces which creates confusion, see
+   `<https://docs.python.org/2/howto/doanddont.html#language-constructs-you-should-not-use>`_.
+   Keeping the original namespace ensures naming consistency in the codebase
+   and speeds up the code reviews: co-developpers and maintainers do not have
+   to check if you are using the original module's method or if you have
+   overwritten it.
+
+9. Use double quotes " and not single quotes ' for strings.
+
+10. If you need several lines for a function call,
```

Comment:
this guidelines may have became outdated with the use of `black`.

Reply:
I suggest we handle this in another PR, to avoid scope creep.